### PR TITLE
[e2e] Rollback node affinity for BYOH label

### PR DESCRIPTION
**Pull request by jrvaldes**
This commit revert the node affinity logic introduced, in order to keep
the e2e-upgrade test low in complexity.

To ensure the presence of a running workload in a BYOH node, if any, we trust
the kube-scheduler placement algorithm, that balance the number of workloads
across the available nodes that match taints and selectors.

The number of workloads is set to 3, representing the total number of
nodes (one BYOH and one WM) + 1, assuring the eviction of at least 1 pod from
the BYOH node during the de-configuration process.

### CODE REVIEW

File: pkg/nodeconfig/nodeconfig.go [hunk 1]
```diff
@@ -77,8 +77,8 @@ type ErrWriter struct {
 }
 
 func (ow OutWriter) Write(p []byte) (n int, err error) {
-	// log debug
-	ow.log.V(1).Info(string(p))
+	// log info
+	ow.log.Info(string(p))
```
Comment (jrvaldes) [inline pkg/nodeconfig/nodeconfig.go:81]: Sample output below. There is only a minor cosmetic issue, it adds a [line break](https://github.com/openshift/windows-machine-config-operator/blob/master/vendor/k8s.io/kubectl/pkg/drain/drain.go#L292).

```
	INFO	controllers.configmap	processing	{"instances in": "windows-instances"}
	DEBUG	wc 172.31.251.202	initializing SSH connection
	INFO	nc 172.31.251.202	evicting pod default/win-webserver-wm-6756f7468f-scbm8

	INFO	nc 172.31.251.202	evicting pod default/win-webserver-wm-6756f7468f-zmnc4


```

File: test/e2e/upgrade_test.go [hunk 2]
```diff
@@ -73,27 +52,20 @@ func upgradeTestSuite(t *testing.T) {
...
+
+	// create a clusterIP service which can be used to reach the Windows webserver
+	intermediarySVC, err := testCtx.createService(deployment.Name, v1.ServiceTypeClusterIP, *deployment.Spec.Selector)
+	require.NoErrorf(t, err, "error creating service for deployment %s", deployment.Name)
+	defer testCtx.deleteService(intermediarySVC.Name)
+
+	// create a Job object that continuously curls the webserver every 5 seconds.
+	testerJob, err := testCtx.createLinuxJob(windowsWorkloadTesterJob, generateLinuxWorkloadTesterCommand(intermediarySVC.Spec.ClusterIP))
+	require.NoErrorf(t, err, "error creating linux job %s", windowsWorkloadTesterJob)
+	defer testCtx.deleteJob(testerJob.Name)
```
Comment (aravindhp) [inline test/e2e/upgrade_test.go:68]: Please revert back to how the code was before i.e. reintroduce `deployWindowsWorkloadAndTester()`. Sorry this is not concise code but belongs to a method that by looking at the signature one can understand what it does. I had called this out [earlier](https://github.com/openshift/windows-machine-config-operator/pull/645#discussion_r710261878).

**Review by aravindhp — CHANGES_REQUESTED** [review]

File: test/e2e/upgrade_test.go (hunk 2 above)
Comment (jrvaldes) [reply test/e2e/upgrade_test.go]: Reverted in https://github.com/openshift/windows-machine-config-operator/pull/674/commits/d3b3c0c2bdc42999910ff263994ae8516d85044e

**Review by saifshaikh48 — COMMENTED** [review]
/lgtm

File: test/e2e/upgrade_test.go [hunk 3]
```diff
@@ -206,3 +168,31 @@ func (tc *testContext) scaleWMCODeployment(desiredReplicas int32) error {
...
+	testerJob, err := tc.createLinuxJob(windowsWorkloadTesterJob, generateLinuxWorkloadTesterCommand(intermediarySVC.Spec.ClusterIP))
+	if err != nil {
+		return nil, errors.Wrapf(err, "error creating linux job %s", windowsWorkloadTesterJob)
+	}
+	// return a cleanup func
+	return func() {
+		// ignore errors while deleting the objects
+		_ = tc.deleteDeployment(deployment.Name)
+		_ = tc.deleteService(intermediarySVC.Name)
+		_ = tc.deleteJob(testerJob.Name)
```
Comment (aravindhp) [inline test/e2e/upgrade_test.go:196]: Do you really need `_ =`? It should be sufficient to leave that out given you have added the comment.

File: test/e2e/upgrade_test.go [hunk 4]
```diff
@@ -206,3 +168,31 @@ func (tc *testContext) scaleWMCODeployment(desiredReplicas int32) error {
...
+func (tc *testContext) deployWindowsWorkloadAndTester() (func(), error) {
+	// create a Windows Webserver deployment
+	deployment, err := tc.deployWindowsWebServer("win-webserver", nil)
+	if err != nil {
+		return nil, errors.Wrap(err, "error creating Windows Webserver deployment for upgrade test")
+	}
+	// create a clusterIP service which can be used to reach the Windows webserver
+	intermediarySVC, err := tc.createService(deployment.Name, v1.ServiceTypeClusterIP, *deployment.Spec.Selector)
+	if err != nil {
+		return nil, errors.Wrapf(err, "error creating service for deployment %s", deployment.Name)
```
Comment (aravindhp) [inline test/e2e/upgrade_test.go:184]: By return `nil` here you will end up not deleting the deployment and leak resources. How are you going to handle that?
Yes, it will get cleaned up on cluster deletion but given we are attempting a clean up, we should do that consistently. If you are not doing so for any reason, then it needs to be called here in a comment.

File: test/e2e/upgrade_test.go [hunk 5]
```diff
@@ -206,3 +168,31 @@ func (tc *testContext) scaleWMCODeployment(desiredReplicas int32) error {
...
+	}
+	// create a Job object that continuously curls the webserver every 5 seconds.
+	testerJob, err := tc.createLinuxJob(windowsWorkloadTesterJob, generateLinuxWorkloadTesterCommand(intermediarySVC.Spec.ClusterIP))
+	if err != nil {
+		return nil, errors.Wrapf(err, "error creating linux job %s", windowsWorkloadTesterJob)
```
Comment (aravindhp) [inline test/e2e/upgrade_test.go:189]: Same comment here.

**Review by aravindhp — CHANGES_REQUESTED** [review]

File: test/e2e/upgrade_test.go (hunk 3 above)
Comment (jrvaldes) [reply test/e2e/upgrade_test.go:196]: It can be removed, yes; it's a matter of style. `_ =`  is the `go` way of saying  "explicitly ignore the returned error", and the inline comment complements that.

File: test/e2e/upgrade_test.go (hunk 4 above)
Comment (jrvaldes) [reply test/e2e/upgrade_test.go:184]: Good catch, we should do that consistently and cleanup the already created objects. I missed that edge case, will adjust.

File: test/e2e/upgrade_test.go (hunk 5 above)
Comment (jrvaldes) [reply test/e2e/upgrade_test.go:192]: Fixed in https://github.com/openshift/windows-machine-config-operator/pull/674/commits/267e616e832d92bbae86788c2612b4224a78d5a8

File: test/e2e/upgrade_test.go [hunk 6]
```diff
@@ -206,3 +168,34 @@ func (tc *testContext) scaleWMCODeployment(desiredReplicas int32) error {
...
+		return nil, errors.Wrap(err, "error creating Windows Webserver deployment for upgrade test")
+	}
+	// create a clusterIP service which can be used to reach the Windows webserver
+	intermediarySVC, err := tc.createService(deployment.Name, v1.ServiceTypeClusterIP, *deployment.Spec.Selector)
+	if err != nil {
+		_ = tc.deleteDeployment(deployment.Name)
+		return nil, errors.Wrapf(err, "error creating service for deployment %s", deployment.Name)
+	}
+	// create a Job object that continuously curls the webserver every 5 seconds.
+	testerJob, err := tc.createLinuxJob(windowsWorkloadTesterJob, generateLinuxWorkloadTesterCommand(intermediarySVC.Spec.ClusterIP))
```
Comment (aravindhp) [inline test/e2e/upgrade_test.go:188]: This line goes beyond 120 columns

**Review by aravindhp — APPROVED** [review]
/approve

with the caveat you fix the long line.

File: test/e2e/upgrade_test.go (hunk 6 above)
Comment (jrvaldes) [reply test/e2e/upgrade_test.go]: Fixed.
